### ISSUE TITLE
[Bug] Plugin doesn't detect problems using `this.page` when using page object model

### ISSUE THREAD

**Issue by dwick** [playwright-community/eslint-plugin-playwright#112]
Hi, this plugin looks really useful however it is missing a lot of errors when using the ["page object model"](https://playwright.dev/docs/pom)
```typescript
import { Page } from '@playwright/test';

export default class TestPage {
  protected readonly page: Page;
  constructor(page: Page) {
    this.page = page;
  }

  async testNoPagePause() {
    // no error
    await this.page.pause();

    const page = this.page;
    // Unexpected use of page.pause(). eslint(playwright/no-page-pause)
    await page.pause();
  }

  async testNoElementHandles() {
    // no error
    await expect(this.page.$('button')).toBeVisible();

    const page = this.page;
    // Unexpected use of element handles. eslint(playwright/no-element-handle)
    await expect(page.$('button')).toBeVisible();
  }

  async testNoWaitForTimeout() {
    // no error
    await this.page.waitForTimeout(5000);

    const page = this.page;
    // Unexpected use of page.waitForTimeout(). eslint(playwright/no-wait-for-timeout)
    await page.waitForTimeout(5000);
  }
}
```

Not sure how feasible this is with eslint, but it would be super awesome to get it working.

**Comment by mskelton** [comment]
Yeah, this should be feasible to fix, just need to adjust how we are recognizing `page` variables.

**Comment by GuyMograbi** [comment]
Yes please fix it. 
I just plugged this into my eslint and was very disappointed to see this.

**Comment by mskelton** [comment]
@GuyMograbi You're welcome to contribute PRs.  I haven't had much time for this plugin as of late.

**Comment by GuyMograbi** [comment]
@mskelton thank you so much for resolving this. 

I actually tried to have a stab at it.. but I would not have gotten as good a result as you. Learned a lot from your PR. 

--
Is it possible to release a version with this fix? 
Can I just open a PR to bump the version?

**Comment by mskelton** [comment]
@GuyMograbi I'm going to get a few other features in the next release, most notably #113 as well as a few other minor improvements.  Hoping to release this week.
